api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/TxTbPrevQueriesInterface.java

```diff
@@ -0,0 +1,179 @@
+            + "inner join encounter e on p.patient_id=e.patient_id "
+            + "inner join obs o on o.encounter_id=e.encounter_id "
+            + "where e.voided=0 and p.voided=0 and e.encounter_datetime between :startDate - interval 6 month and :endDate - interval 6 month and "
+            + "o.voided=0 and o.concept_id=6122 and o.value_coded=1256 and e.encounter_type=6 and  e.location_id=:location group by p.patient_id) startTPI group by patient_id) inicio_tpi "
+            + "inner join encounter e on e.patient_id=inicio_tpi.patient_id "
+            + "inner join obs o on o.encounter_id=e.encounter_id "
+            + "where e.encounter_type in (6,9) and e.voided=0 and e.encounter_datetime between data_inicio_tpi and data_inicio_tpi + interval 7 month and "
+            + "o.concept_id=6122 and o.value_coded in (1257,1065) and o.voided=0 and e.location_id=:location "
+            + "group by inicio_tpi.patient_id having count(*)>=6 ";
+
+    public static final String findPatientsTransferredOut =
+        "select transferidopara.patient_id from ( "
+            + "select patient_id,max(data_transferidopara) data_transferidopara from ( "
+            + "select pg.patient_id,max(ps.start_date) data_transferidopara from patient p "
+            + "inner join patient_program pg on p.patient_id=pg.patient_id "
+            + "inner join patient_state ps on pg.patient_program_id=ps.patient_program_id "
+            + "where pg.voided=0 and ps.voided=0 and p.voided=0 and  pg.program_id=2 and ps.state=7 and ps.end_date is null and  ps.start_date<=:endDate and location_id=:location group by p.patient_id "
+            + "union "
+            + "select p.patient_id,max(e.encounter_datetime) data_transferidopara from patient p "
+            + "inner join encounter e on p.patient_id=e.patient_id "
+            + "inner join obs o on o.encounter_id=e.encounter_id where 	e.voided=0 and p.voided=0 and e.encounter_datetime<=:endDate and "
+            + "o.voided=0 and o.concept_id=2016 and o.value_coded in (1706,23863) and e.encounter_type=21 and  e.location_id=:location group by p.patient_id "
+            + "union "
+            + "select p.patient_id,max(o.obs_datetime) data_transferidopara from patient p "
+            + "inner join encounter e on p.patient_id=e.patient_id "
+            + "inner join obs o on o.encounter_id=e.encounter_id "
+            + "where 	e.voided=0 and p.voided=0 and o.obs_datetime<=:endDate and "
+            + "o.voided=0 and o.concept_id=6272 and o.value_coded=1706 and e.encounter_type=53 and  e.location_id=:location group by p.patient_id "
+            + "union "
+            + "select p.patient_id,max(e.encounter_datetime) data_transferidopara from	patient p "
+            + "inner join encounter e on p.patient_id=e.patient_id "
+            + "inner join obs o on o.encounter_id=e.encounter_id "
+            + "where e.voided=0 and p.voided=0 and e.encounter_datetime<=:endDate and "
```

Comment:
@emaposse, the same for **Ficha Clinica**

ps.start_date>=:**startDate – 6 months** and ps.start_date<=:endDate

Reply:
Fixed @Guimino.